pkg/reconciler/route/reconcile_resources.go

```diff
@@ -173,30 +215,86 @@ func (c *Reconciler) reconcilePlaceholderServices(ctx context.Context, route *v1
 	return services, nil
 }
 
-func (c *Reconciler) updatePlaceholderServices(ctx context.Context, route *v1.Route, services []*corev1.Service, ingress *netv1alpha1.Ingress) error {
+func (c *Reconciler) updatePlaceholderServices(ctx context.Context, route *v1.Route, pairs []resources.ServicePair, ingress *netv1alpha1.Ingress) error {
 	logger := logging.FromContext(ctx)
 	ns := route.Namespace
 
 	eg, egCtx := errgroup.WithContext(ctx)
-	for _, service := range services {
-		service := service
+	for _, pair := range pairs {
+		from := pair
 		eg.Go(func() error {
-			desiredService, err := resources.MakeK8sService(egCtx, route, service.Name, ingress, resources.IsClusterLocalService(service), service.Spec.ClusterIP)
+			to, err := resources.MakeK8sService(egCtx, route, from.Tag, ingress, resources.IsClusterLocalService(from.Service))
 			if err != nil {
 				// Loadbalancer not ready, no need to update.
 				logger.Warnw("Failed to update k8s service", zap.Error(err))
 				return nil
 			}
 
-			// Make sure that the service has the proper specification.
-			if !equality.Semantic.DeepEqual(service.Spec, desiredService.Spec) {
-				// Don't modify the informers copy.
-				existing := service.DeepCopy()
-				existing.Spec = desiredService.Spec
-				if _, err := c.kubeclient.CoreV1().Services(ns).Update(ctx, existing, metav1.UpdateOptions{}); err != nil {
+			canUpdate := false
+
+			if from.Spec.Type != to.Spec.Type {
+				switch from.Spec.Type {
+				// Transitions from ExternalName to any type should work
+				case corev1.ServiceTypeExternalName:
+					canUpdate = true
+				default:
+					// Transitions from ClusterIP to ExternalName Fail
```

Comment:
We need to delete the placeholder services manually when downgrading the serving. We probably need to document it?

Reply:
Good point will add this as release note

Reply:
I think this is bad as this could cause downgrading fails by default for all users without any manual work. And I really feel that we should handle the backward compatibility issue instead of just documenting it.

Would it be OK to:
1) putting this feature behind a flag and set the flag to false in 0.26 release.
2) handling the backward compatibility issue in 0.26 release
3) enabling the feature by default in 0.27 release

Reply:
Did you see my comment here: https://github.com/knative/serving/pull/11843#issuecomment-904824983 about a potential downgrade path that doesn't require manually deleting k8s services?

Also the prior behaviour is preserved - it's only until the `net-*` plugins start setting the LoadBalancers status IP that triggers the changes in this PR. So if net-istio/kourier don't play on setting that value then downgrade will work.

Reply:
Thanks for the pointer, David. The comment makes sense.

For downgrading, I am not sure how easy it would be for users to do the downgrade based on the orders mentioned in the comment. To make the downgrade easier, I would suggest we ship this PR in 0.26 release, and populate Kingress IP from the `net-*` repo in 0.27+ release if the maintainers are OK with it. WDYT @nak3 ?

Reply:
Yes, net-contour needs this change ASAP but net-istio/kourier do not need to rush so adopting it on 0.27+ would be fine. It depends on each repo maintainer's decision as you also said, though.